lib/api-gateway.ts

```diff
@@ -0,0 +1,162 @@
+import apigw = require('@aws-cdk/aws-apigateway');
+import { Metric, MetricOptions, ComparisonOperator, GraphWidget } from '@aws-cdk/aws-cloudwatch';
+import { Construct, Duration } from '@aws-cdk/core';
+import { IWatchful } from './api';
+
+export interface WatchApiGatewayRestApiOptions {
+  /**
+   * Alarm when 5XX errors reach this threshold over 5 minutes.
+   *
+   * @default - no alarm.
+   */
+  readonly serverErrorThreshold?: number;
+
+  /**
+   * A list of operations to monitor separately.
+   *
+   * @default - only API-level monitoring is added.
+   */
+  readonly watchedOperations?: WatchedOperation[];
+}
+
+export interface WatchApiGatewayRestApiProps extends WatchApiGatewayRestApiOptions {
+  /**
+   * The title of this section.
+   */
+  readonly title: string;
+
+  /**
+   * The Watchful instance to add widgets into.
+   */
+  readonly watchful: IWatchful;
+
+  /**
+   * The API Gateway REST API that is being watched.
+   */
+  readonly restApi: apigw.RestApi;
+}
+
+export class WatchApiGatewayRestApi extends Construct {
+  private readonly api: apigw.CfnRestApi;
+  private readonly stage: string;
+  private readonly watchful: IWatchful;
+
+  constructor(scope: Construct, id: string, props: WatchApiGatewayRestApiProps) {
+    super(scope, id);
+
+    this.api = props.restApi.node.findChild('Resource') as apigw.CfnRestApi;
+    this.stage = props.restApi.deploymentStage.stageName;
+    this.watchful = props.watchful;
+
+    if (props.serverErrorThreshold) {
+      this.watchful.addAlarm(
+        this.createApiGatewayMetric(ApiGatewayMetric.FiveHundredError)
+          .createAlarm(this, '5XXErrorAlarm', {
+            alarmDescription: `at ${props.serverErrorThreshold}`,
+            threshold: props.serverErrorThreshold,
+            period: Duration.minutes(5),
+            comparisonOperator: ComparisonOperator.GREATER_THAN_OR_EQUAL_TO_THRESHOLD,
+            evaluationPeriods: 1,
+            statistic: 'sum',
+          })
+      );
+    }
+
+    this.watchful.addSection(props.title, {
+      links: [{ title: 'Amazon API Gateway Console', url: linkForApiGateway(props.restApi) }]
+    });
+    [undefined, ...props.watchedOperations || []].forEach(operation =>
+      this.watchful.addWidgets(
+        this.createCallGraphWidget(operation),
+        this.createCacheGraphWidget(operation),
+        this.createLatencyGraphWidget(ApiGatewayMetric.Latency, operation),
+        this.createLatencyGraphWidget(ApiGatewayMetric.IntegrationLatency, operation),
+      )
+    );
+  }
+
+  private createCallGraphWidget(opts?: WatchedOperation) {
+    return new GraphWidget({
+      title: `${opts ? `${opts.httpMethod} ${opts.resourcePath}` : 'Overall'} Calls`,
```

Comment:
Add period in the title (see other resources, which I've just added)

Reply:
The pattern seems to make sense for sum aggregates (Counts / N minutes), however how'd you see this for non-sum aggregates (e.g: P90 latency)?

Reply:
Not sure I understand, it's still accumulated over a certain period, isn't it?